src/puya/awst_build/subroutine.py

```diff
@@ -611,6 +616,14 @@ def _visit_ref_expr_maybe_aliased(
         if fullname.startswith(constants.PUYAPY_PREFIX):
             return self._visit_ref_expr_of_puyapy(fullname, expr_loc, expr.node)
         match expr:
+            case mypy.nodes.RefExpr(node=mypy.nodes.TypeInfo() as typ) | mypy.nodes.NameExpr(
```

Comment:
the ` | NameExpr` shouldn't be required, since it's a subtype of `RefExpr` and you're capturing the same variables either way?

Reply:
separately, an existing issue is that we're not correctly handling explicit base calls like `MyBase.some_method(self, ...)`, which we should - but this code here would capture those references